mula/scheduler/schedulers/boefje.py

```diff
@@ -997,3 +997,29 @@ def get_boefjes_for_ooi(self, ooi) -> list[Plugin]:
         )
 
         return boefjes
+
+    def set_cron(self, item: Task) -> str | None:
+        """Override Schedule.set_cron() when a boefje specifies a schedule for
+        execution (cron expression) we schedule for its execution"""
+        # Does a boefje have a schedule defined?
+        plugin = self.ctx.services.katalogus.get_plugin_by_id_and_org_id(
+            utils.deep_get(item.data, ["boefje", "id"]),
+            self.organisation.id,
+        )
+        if plugin is None or plugin.cron is None:
+            return super().set_cron(item)
+
+        return plugin.cron
+
+    def calculate_deadline(self, task: Task) -> datetime:
+        """Override Scheduler.calculate_deadline() to calculate the deadline
+        for a task and based on the boefje interval."""
+        # Does the boefje have an interval defined?
+        plugin = self.ctx.services.katalogus.get_plugin_by_id_and_org_id(
```

Comment:
Dont we already have the plugins available somewhere? Or does this not trigger a call to the Katalogus api?

Reply:
The katalogus service in the scheduler has a cache implemented, it should not trigger a call when it isn't necessary